src/map/script.c

```diff
@@ -12618,6 +12622,47 @@ BUILDIN(waitingroomkickall) {
 	return true;
 }
 
+/// Kicks target player from the waiting room of the current or target npc.
+///
+/// kickwaitingroom "<npc_name>"{,"<name>"|<account id>};
+/// kickwaitingroom;
+BUILDIN(waitingroomkick) {
+	struct npc_data* nd;
+	struct chat_data* cd;
+	struct map_session_data *sd = NULL;
+
+	if (script_hasdata(st, 2)) {
+		nd = npc->name2id(script_getstr(st, 2));
+		if (nd == NULL) {
+			ShowWarning("buildin_waitingroomkick: NPC name '%s' not found.\n", script_getstr(st, 2));
+			return false;
+		}
+	}
+	else {
+		nd = map->id2nd(st->oid);
+	}
+
+	if (nd != NULL && (cd = map->id2cd(nd->chat_id)) != NULL) {
+		if (script_hasdata(st, 3)) {
+			if (script_isstringtype(st, 3)) {
+				sd = map->nick2sd(script_getstr(st, 3));
+			}
+			else {
+				sd = map->id2sd(script_getnum(st, 3));
+			}
+
+			if (sd != NULL && sd->chat_id != 0) {
+				chat->leave(sd, false);
+			}
```

Comment:
see [coding guidelines chapter 3](https://github.com/HerculesWS/Hercules/wiki/Coding-Style#chapter-3-placing-braces-and-spaces)
>Do not unnecessarily use braces where a single statement will do.

	if (condition)
		action();

and

	if (condition)
		do_this();
	else
		do_that();